src/OrchardCore/OrchardCore.Recipes.Abstractions/Models/RecipeExecutionException.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+
+namespace OrchardCore.Recipes.Models;
+
+public class RecipeExecutionException : Exception
+{
+    public RecipeStepResult StepResult { get; }
+
+    public RecipeExecutionException(RecipeStepResult recipeStepResult)
+    {
+        StepResult = recipeStepResult;
+    }
+
+    public RecipeExecutionException(Exception exception, RecipeStepResult recipeStepResult)
+        : base(exception.Message, exception)
+    {
+        StepResult = recipeStepResult;
+    }
+}
```

Comment:
```suggestion
public class RecipeExecutionException : Exception
{
    public RecipeExecutionException(RecipeStepResult recipeStepResult)
    {
        StepResult = recipeStepResult;
    }

    public RecipeExecutionException(Exception exception, RecipeStepResult recipeStepResult)
        : base(exception.Message, exception)
    {
        StepResult = recipeStepResult;
    }
    
    public RecipeStepResult StepResult { init; get; }
}
```

Reply:
What changed here?

Reply:
- Move the property after the constructor
- Use init get property

Reply:
Naaah, I like how I had it. No need for init either

Reply:
If you dislike `init` properties should come after the constructor in the guidelines

Reply:
What guideline? Why?

Reply:
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1201.md

Reply:
FYI this not specific for this PR, but I'm trying to make things organized as much as I can :)

Reply:
I don't think we enforce it. By any way we changed it.